Automattic/woocommerce-payments
# Add compatibility for Multi-Currency and UPS plugin.

Fixes #3132 

#### Changes proposed in this Pull Request

The format of the new compatibility class(es) is based off of the work @tpaksu [has done in the Deposits PR](https://github.com/Automattic/woocommerce-payments/pull/3041), and was discussed in Slack: p1634314277040500-slack-C01B8KNUYSW

* Define a new constant, `FILTER_PREFIX` in the `Compatibility` class to allow for use in filters as this grows and gets refactored.
* New `should_return_store_currency` method in `Compatibility`, which is used in `FrontendCurrencies` to determine if the store or selected currency should be used. This method returns the results of a filter so that we can move functionality to separate classes.
* New `WooCommerceUPS` class in the `WCPay\MultiCurrency\Compatibility` namespace.
  * Uses the `wcpay_multi_currency_should_return_store_currency` to determine whether to return the store currency or the selected currency.
* Add `Compatibility` into `FrontendCurrencies` to get this all to work.
* Add or update tests as needed.

#### Testing instructions

* You will need to install the UPS plugin and set it up.
  * It can be found in the `all-plugins` repo.
  * The creds for setup can be found in the secret store, and they go under WooCommerce > Settings > Shipping > UPS.
  * If you haven't set up shipping before, you can add a new shipping zone, and if you don't specify a _Zone region_, it works for everywhere.
  * Once UPS is added to your shipping zone, you can set the Origin City, Postcode, and Country to Ocala, 34480, United States - Florida. (It's what I use) https://d.pr/i/RJfRtS
  * Make sure all _Services_ are enabled. https://d.pr/i/s7FQ1o
  * That should be it.
* You will need to have a product that has dimensions and a weight set on it. I typically do 5x5x5 for dimensions and a weight of 5.
* Add the item to your cart, and then go to the cart page.
* Check your shipping costs in your store currency against other currencies to make sure to prices are converted correctly.

-------------------


**Post merge**
